### PR TITLE
Integrate Danger in CI

### PR DESCRIPTION
**Pull request by lacerda-eduardo**
### Checklist
- [x] I've rebased my changes on top of `master`
- [x] I've built and run the project to see all new and existing tests pass
- [x] I've followed the [Mindera swift style guide](https://github.com/Mindera/swift-style-guide)
- [x] I've read the [Contribution Guidelines](https://github.com/Mindera/Alicerce/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Install Danger-Swift (https://danger.systems/swift/) to automate common code reviews rules. Danger will write comments on PR as 'github-actions' bot.
Danger is installed just to run on CI, it not a project dependency.

### Description
- Add danger on brew file to be installed at CI.
- Create Dangerfile.swift, at this file is configured all rules that Danger will check on PRs. For now we just have SwiftLint rules and a List of all changed files. 
- Change ci.yml to add Danger step. It runs 417-72KI/danger-swiftlint@v3 GitHub Action.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -112,6 +112,16 @@ jobs:
       - name: carthage build
         run: ./script/carthage.sh build --cache-builds --no-skip-current
 
+  Danger:
+    name: Danger Swift
+    runs-on: ubuntu-latest
```
Comment (diogoeiras) [inline .github/workflows/ci.yml:117]: Maybe lock to the specific version as we do for Mac? `ubuntu-20.04`

File: Dangerfile.swift [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import Danger
+
+let danger = Danger()
+
+// Swiftlint style check, with comment on PR
+SwiftLint.lint(.all(directory: nil), inline: true)
+
+// List of all edited files
+let editedFiles = danger.git.modifiedFiles + danger.git.createdFiles
+message("These files have changed: \(editedFiles.joined(separator: ", "))")
```
Comment (diogoeiras) [inline Dangerfile.swift:12]: Do we need to list files changed? What do we gain with this? Also, we're missing `danger.git.deletedFiles`

**Review by diogoeiras — APPROVED** [review]

File: Dangerfile.swift (hunk 2 above)
Comment (lacerda-eduardo) [reply Dangerfile.swift]: @diogoeiras I removed this rules (was using for test). Keep only SwiftLint until we have more rules to this specific project.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -112,6 +112,16 @@ jobs:
       - name: carthage build
         run: ./script/carthage.sh build --cache-builds --no-skip-current
 
+  Danger:
```
Comment (p4checo) [inline .github/workflows/ci.yml:115]: I think this could live in a separate action instead, like `danger.yml` which only runs on PRs

File: Brewfile [hunk 4]
```diff
@@ -1 +1,3 @@
 brew 'Swiftlint'
+tap "danger/tap"
+brew "danger-swift"
```
Comment (p4checo) [inline Brewfile:3]: are these changes still needed, given Danger is provided by the `417-72KI/danger-swiftlint@v3` action?

File: Dangerfile.swift [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// Copyright © 2021 Mindera. All rights reserved.
```
Comment (p4checo) [inline Dangerfile.swift:1]: I think we don't need this copyright notice

**Review by p4checo — CHANGES_REQUESTED** [review]
Nice work! Still think we can polish it a bit though 🤓

File: Brewfile (hunk 4 above)
Comment (lacerda-eduardo) [reply Brewfile]: @p4checo do you think we should keep Swiflint? It is now integrated with Danger, maybe we could remove it too.

Comment (p4checo) [reply Brewfile]: I think we can. 

It was added here during the initial SwiftLint integration in the project, but funnily enough it was never invoked via CI 😅

We also run SwiftLint locally in the project, which is done via a SwiftLint target that contains a shell script to run `swiftlint` and print a warning if it isn't installed. In hindsight, this script should trigger an error instead of just printing... 🤔

File: Dangerfile.swift [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import Danger
+
+let danger = Danger()
+
+// Swiftlint style check, with comment on PR
+SwiftLint.lint(.all(directory: nil), inline: true)
```
Comment (p4checo) [inline Dangerfile.swift:6]: perhaps add a trailing newline?

File: .github/workflows/danger-ci.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+name: Danger-CI
```
Comment (p4checo) [inline .github/workflows/danger-ci.yml:1]: perhaps just `Danger` 🤓 💅🏼

File: Dangerfile.swift [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
```
[the remark this answers is not in the document]
Comment (p4checo) [reply Dangerfile.swift:7]: I don't get why SwiftLint is complaining here 🤔

[the remark this answers is not in the document]
Comment (lacerda-eduardo) [reply Dangerfile.swift:7]: Me neither.

**Review by p4checo — APPROVED** [review]
🚀

**Review by filipe-lemos — APPROVED** [review]
